### ISSUE TITLE
kubectl bash file

### ISSUE THREAD

**Issue by diskmanti** [jhunt/env#3]
Hello I stumbled upon your posts on [kubectl tricks](https://starkandwayne.com/silly-kubectl-tricks/) and from there i found this repo with all your configs. I was wondering what did you do on /bash/kubectl config.

Also I think the Chrono Trigger characters pixel art on your blog looks very cool 

Cheers

**Comment by jhunt** [comment]
The [bash/kubectl](https://github.com/jhunt/env/blob/master/bash/kubectl) file is just a dump of Kubernetes completions, IIRC.  The real magic is in [bin/k](https://github.com/jhunt/env/blob/master/bin/k), which gives me a few niceties:

- `k` is a global alias of `kubectl`, regardless of shell
- `k` can be exec'd by things like `watch k get pods`, which I do _a lot_.
- `k` has some logic in it to find alternate names for `kubectl`, like `microk8s.kubectl`, mainly so that I can forget that some distributions of k8s rename the binary in the first place.

Other than that, I have [bin/kid](https://github.com/jhunt/env/blob/master/bin/kid) (which I talk about in one of the posts / blogs, and not a whole heck of a lot else (utility-wise).

One thing I absolutely cannot live without these days is my prompt, which I stuffed the current context and namespace into:

<img width="757" alt="Screen Shot 2020-04-27 at 10 03 26 PM" src="https://user-images.githubusercontent.com/20047/80438804-0fa26780-88d3-11ea-93eb-097ca84349a8.png">

That way, I can tell at a glance that (at least in this screenshot) I'm using a cluster named `arcades` in the `vcb` namespace.  I tend to put a lot of stuff in my prompt but this was a massive boon.

I used to do `k is` to set `$KUBECONFIG` in a primitive (and quite brittle) form of `kubectx`, but I've since stopped doing that and just munged all of my KUBECONFIGs together into one big `~/.kube/config` file that `kubectx` and `kubens` can use.
